WIP on KV pushdown

yuzefovich opened the pull request:
Epic: None

yuzefovich committed:
colfetcher: propagate TraceKV and size hints for the KV pushdown

Release note: None

yuzefovich committed:
WIP on disabling

yuzefovich committed:
WIP on enabling

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/98263)
<!-- Reviewable:end -->

yuzefovich committed:
colfetcher: track KV CPU time in the direct columnar scan

This commit addresses a minor TODO to track the KV CPU time when direct
columnar scans are used. In the regular columnar scan this time is
tracked by the cFetcher, but with the KV projection pushdown the
cFetcher is used on the KV server side, so we need to augment the
ColBatchDirectScan to track it. Notably, this means that the decoding
done on the KV server side is included. Additionally, this commit
clarifies how the KV CPU time is obtained from the cFetcher (we don't
need to use a helper (unlike in the case of `bytesRead` and
`batchRequestsIssued` fields which are written to on `cFetcher.Close`)).

Release note: None

yuzefovich committed:
WIP on skipping BenchmarkSQL
